core/app/services/spree/credit_cards/destroy.rb

```diff
@@ -0,0 +1,18 @@
+module Spree
+  module CreditCards
+    class Destroy
+      prepend Spree::ServiceModule::Base
+
+      def call(card:)
+        ActiveRecord::Base.transaction do
+          card.payments.
```

Comment:
@aplegatt 

`invalidate!` works only on payments in state `checkout`
for `pending` and `processing` states you need to call `void!`

please see: https://github.com/spree/spree/blob/master/core/app/models/spree/payment.rb#L77